Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
```

Comment:
We're adding it as we want to request `EmailAddressLocalPart` in the `MailAddressGenerator` instead of requesting a string because we want the mail address specimen to be in conformity with the SMTP format and we can't guarantee the logic for string generation hasn't been modified to be incompatible with that format.  That seems like an argument for having `EmailAddressLocalPart` perform validation on the local part although I could be completely missing what you're driving at.